### ISSUE TITLE
Linking problems on OS X with kf/gallium (0.5)

### ISSUE THREAD

**Issue by iamed2** [Keno/Cxx.jl#200]
```julia
julia> Pkg.build("Cxx")
INFO: Building Cxx
Tuning for julia installation at: /Users/ericdavies/repos/juliagallium/usr/bin
    CC /Users/ericdavies/.julia/juliagallium/v0.5/Cxx/deps/build/bootstrap.o
    LINK /Users/ericdavies/.julia/juliagallium/v0.5/Cxx/deps/usr/lib/libcxxffi.dylib
Undefined symbols for architecture x86_64:
  "clang::CodeGen::CGBuilderInserter<false>::InsertHelper(llvm::Instruction*, llvm::Twine const&, llvm::BasicBlock*, llvm::ilist_iterator<llvm::Instruction>) const", referenced from:
      llvm::CastInst* llvm::IRBuilder<false, llvm::ConstantFolder, clang::CodeGen::CGBuilderInserter<false> >::Insert<llvm::CastInst>(llvm::CastInst*, llvm::Twine const&) const in bootstrap.o
      llvm::ExtractValueInst* llvm::IRBuilder<false, llvm::ConstantFolder, clang::CodeGen::CGBuilderInserter<false> >::Insert<llvm::ExtractValueInst>(llvm::ExtractValueInst*, llvm::Twine const&) const in bootstrap.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [usr/lib/libcxxffi.dylib] Error 1
```

Julia built with this `Make.user`:
```make
override LLDB_VER=master
override LLVM_VER=svn
override LLVM_ASSERTIONS=1
override BUILD_LLVM_CLANG=1
override BUILD_LLDB=1
override USE_LLVM_SHLIB=1
override LLDB_DISABLE_PYTHON=1

override LLVM_GIT_URL_LLVM=https://github.com/JuliaLang/llvm.git
override LLVM_GIT_URL_LLDB=https://github.com/JuliaLang/lldb.git
override LLVM_GIT_URL_CLANG=https://github.com/JuliaLang/clang.git
override LLVM_GIT_VER=kf/gallium
override LLVM_GIT_VER_LLDB=kf/gallium
override LLVM_GIT_VER_CLANG=kf/gallium

OPENBLAS_TARGET_ARCH=HASWELL
```

**Comment by Keno** [comment]
Do you have https://github.com/Keno/Cxx.jl/commit/3b6d218b98c32038ff9d920944285945802ad28a?

**Comment by iamed2** [comment]
Yup, HEAD is at 748cc1a79eff45887714801f7645c778483d19d2

**Comment by andreasnoack** [comment]
I also get this error on my Mac with a similar setup.

**Comment by Keno** [comment]
Fixed in d03998716d780cc92a2f491ebbd88790c0421fea, but you might want to just cherry-pick that until I get around to rebasing kf/gallium, because I also pushed some changes for Clang API changes recently.

**Comment by iamed2** [comment]
Cherry-picked and it worked, thanks
